bvm/ballerina-runtime/README.md

```diff
@@ -0,0 +1,256 @@
+```
+
+## Create a Ballerina type
+
+`io.ballerina.runtime.api.creators.TypeCreator` class provides APIs to create Ballerina type instances. There are multiple methods available to create Ballerina type instances based on the requirements.
+
+For example, a string array type can be created as follows,
+
+```
+ArrayType strArrayType = TypeCreator.createArrayType(PredefinedTypes.TYPE\_STRING)
+```
+
+## Calling ballerina object method
+
+Ballerina runtime has exposed APIs to call ballerina object method and ballerina function pointer using Java.
+
+io.ballerina.runtime.api.Runtime class exposes two java APis to call object methods.
+
+1. invokeMethodAsyncSequentially
+
+Invoke Object method asynchronously and sequentially. This method will ensure that the object methods are invoked in the same thread where other object methods are executed. So, the methods will be executed sequentially per object level.
+
+2. invokeMethodAsyncConcurrently
+
+Invoke Object method asynchronously and concurrently. Caller needs to ensure that no data race is possible for the mutable state with a given object method and with arguments. So, the method can be concurrently run with different os threads.
+
+Ex. call an isolated method using Java API.
+
+#### Ballerina
+
+```ballerina
+
+import ballerina/jballerina.java;
+public class Person {
+
+    public string name;
+    public int age;
+    
+    function init(string name, int age) {
+        self.name = name;
+        self.age = age;
+    }
+
+    public function play(string sport = "cricket") returns string {
+        return sport;
+    }
+        
+    public function callPlayWithArgs(string s) returns string = @java:Method {
+        'class: "org.ballerinalang.examples.Test"
+    } external;
+}
+
+public function main() {
+    Person p = new Person("John", 30);
+    string sport = p.callPlay("football");
+}
+```
+
+#### Java
+```Java
+
+class Test {
+    public static BString callPlay(Environment env,BObject object,BString bString) {
+        Future future=env.markAsync();
+        env.getRuntime().invokeMethodAsyncConcurrently(object,"play","play",null,
+            new Callback(){
+                @Override
+                public void notifySuccess(Object result){
+                    future.complete(result);
+                }
+    
+                @Override
+                public void notifyFailure(BError error){
+                    future.complete(error);
+                }
+            },null,PredefinedTypes.TYPE_STRING,bString,true);
+            return null;
+        }
+}
+```
+
+
+**Note:** If caller can ensure that given object and object method is isolated and no data race is possible for the mutable state with given arguments, use @invokeMethodAsyncConcurrently
+ otherwise @invokeMethodAsyncSequentially . We can decide the object method isolation if and only if both object.getType().isIsolated() and object.getType().isIsolated(methodName) returns true.
```

Comment:
```suggestion
 otherwise, use @invokeMethodAsyncSequentially. We can decide the object method isolation if and only if both object.getType().isIsolated() and object.getType().isIsolated(methodName) returns true.
```